### PR TITLE
[ci/release] Overcomission clusters to avoid straggling nodes

### PR DESCRIPTION
**Pull request by krfricke**
Signed-off-by: Kai Fricke <kai@anyscale.com>

## Why are these changes needed?

We currently spend a lot of time waiting for straggler nodes to come up. If we overcommission our clusters, we may be able to run tests much sooner and save costs.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by rkooo567 — APPROVED** [review]
Let's see if they will run faster...

**Comment by krfricke** [comment]
```
HTTP response body: {"error":{"detail":"Max workers must not be greater than 250"}}
```

Can we reduce the test to only require 240 nodes?

**Review by rkooo567 — COMMENTED** [review]
Can you actually try running with 240 nodes and compare the runtime to the master? I think the only concern is we are detecting some regression because the tests start tasks/actors as many as num_cpus. If we understand the perf change by this, it will be oaky

File: release/nightly_tests/stress_tests/k8s_stress_test_threaded_actor_compute.yaml [hunk 1]
```diff
@@ -10,8 +10,8 @@ head_node_type:
 worker_node_types:
    - name: worker_node
      instance_type: m5.4xlarge
-     min_workers: 200
-     max_workers: 200
+     min_workers: 210
```
Comment (krfricke) [inline release/nightly_tests/stress_tests/k8s_stress_test_threaded_actor_compute.yaml:13]: This file isn't used anywhere

File: release/release_tests.yaml [hunk 2]
```diff
@@ -3759,7 +3759,7 @@
     timeout: 7200
     script: python many_nodes_tests/actor_test.py
     wait_for_nodes:
-      num_nodes: 251
+      num_nodes: 241
```
Comment (rkooo567) [inline release/release_tests.yaml:3762]: Can we revert this change for now? We will do some discussion this Friday, and decide how to proceed this

File: release/release_tests.yaml [hunk 3]
```diff
@@ -3994,7 +3994,7 @@
     timeout: 3600
     script: python distributed/test_many_tasks.py --num-tasks=1000
     wait_for_nodes:
-      num_nodes: 250
+      num_nodes: 240
```
Comment (rkooo567) [inline release/release_tests.yaml:3997]: same here

**Review by rkooo567 — COMMENTED** [review]
LGTM if 250 node test config is reverted

File: release/nightly_tests/stress_tests/stress_tests_compute.yaml [hunk 4]
```diff
@@ -1,7 +1,7 @@
 cloud_id: {{env["ANYSCALE_CLOUD_ID"]}}
 region: us-west-2
 
-max_workers: 100
+max_workers: 900
```
Comment (rkooo567) [inline release/nightly_tests/stress_tests/stress_tests_compute.yaml:4]: is this expected?
